pkg/provider/gitlab/gitlab.go

```diff
@@ -248,6 +256,10 @@ func (g *Gitlab) ValidateStore(store esv1beta1.GenericStore) error {
 		return fmt.Errorf("projectID cannot be empty")
 	}
 
+	if gitlabSpec.Environment == "" {
```

Comment:
It is optional, isn't it? Validating it like this would make it a forward-incompatible breaking change. From what i've read above in `pkg/provider/gitlab` it seems to be optional.